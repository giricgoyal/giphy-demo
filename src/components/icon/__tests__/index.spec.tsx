import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Icon, { Props } from '../index'

describe('<Icon />', () => {
    let container, getByTestId
    const onClickHander = jest.fn()

    const defaultProps: Props = {
        className: 'some-class',
        'data-testid': 'icon',
        icon: 'cross',
        onClick: onClickHander,
    }

    beforeEach(() => {
        ;({ container, getByTestId } = render(<Icon {...defaultProps} />))
    })

    test('should match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('onClick', () => {
        describe('when clicked', () => {
            beforeEach(() => {
                fireEvent.click(getByTestId('icon'))
            })

            test('should call the onClickHandler', () => {
                expect(onClickHander).toHaveBeenCalled()
            })
        })
    })
})
