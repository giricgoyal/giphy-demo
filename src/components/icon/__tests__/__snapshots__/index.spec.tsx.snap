// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should match snapshot 1`] = `
<div>
  <svg
    class="some-class icon"
    data-testid="icon"
  >
    <use
      xlink:href="#undefined"
    />
  </svg>
</div>
`;
